### PR TITLE
Only feed those retrieved tables to Text2SQL generator via prompt

### PR DESCRIPTION
**Pull request by parkervg**
- Modify `_get_schema` to take a `table_ids: List[str]` argument
- Do some basic normalization on sqlite `CREATE TABLE` statements 
- Only pass those tables returned by retriever to `self.task_generator.generate` call

Below is an example of the new `table_str` value being passed to `self.task_generator.generate`. Notice how, with `k=5`, the 3 tables of the `conference` database that were retrieved are grouped together.

```
Database Name: conference
Schema:

CREATE TABLE "conference" (
	"Conference_ID" int,
	"Conference_Name" text,
	"Year" int,
	"Location" text,
	PRIMARY KEY ("Conference_ID")
)

CREATE TABLE institution (
	"Institution_ID" int,
	"Institution_Name" text,
	"Location" text,
	"Founded" int,
	PRIMARY KEY ("Institution_ID")
)

CREATE TABLE "staff" (
	"staff_ID" int,
	"name" text,
	"Age" int,
	"Nationality" text,
	"Institution_ID" int,
	PRIMARY KEY ("staff_ID"),
	FOREIGN KEY (`Institution_ID`) REFERENCES `institution`(`Institution_ID`)
)

---

Database Name: video_game
Schema:

CREATE TABLE "player" (
	"Player_ID" int,
	"Rank_of_the_year" int,
	"Player_name" text,
	"Position" text,
	"College" text,
	PRIMARY KEY ("Player_ID")
)

---

Database Name: bbc_channels
Schema:

CREATE TABLE "director" (
	"Director_ID" int,
	"Name" text,
	"Age" int,
	PRIMARY KEY ("Director_ID")
)

---
```

### CODE REVIEW

File: target_benchmark/tasks/Text2SQLTask.py [hunk 1]
```diff
@@ -84,22 +85,90 @@ def _get_default_dataset_config(cls) -> Dict[str, DatasetConfigDataModel]:
...
+        table_schemas = cur.execute(f"SELECT name, sql FROM sqlite_schema WHERE type='table' AND name IN ({','.join('?' * len(table_ids))})", table_ids).fetchall()
+        schema_str = f"\nDatabase Name: {db_id}\nSchema:"
+
+        def normalize_schema(schema: str) -> str:
+            # Make sure our commas only come after newlines - not the inverse
+            schema = re.sub(r'\n,', r',\n', schema)
+            # Remove any indentation after the newlines
+            schema = re.sub(r'\n\s+(?=[^\s])', r'\n', schema)
+            # Make sure we have tabs after each newline - just not the last one (with a r-paren after it)
+            schema = re.sub(r'\n(?=[^\)])', r'\n\t', schema)
```
Comment (jixy2012) [inline target_benchmark/tasks/Text2SQLTask.py:166]: curious why this tab after newlines is inserted?

**Review by jixy2012 — COMMENTED** [review]
Hi Parker,

Really nice, straightforward changes with clear comments!
Were you able to run tests to test this updated _get_schema function? If not, I can run on my end with the retrieval results that i have.

**Comment by parkervg** [comment]
> Hi Parker,
> 
> Really nice, straightforward changes with clear comments! Were you able to run tests to test this updated _get_schema function? If not, I can run on my end with the retrieval results that i have.

I will be able to run the tests tomorrow!

File: target_benchmark/tasks/Text2SQLTask.py (hunk 1 above)
Comment (jixy2012) [reply target_benchmark/tasks/Text2SQLTask.py:166]: perfect, thanks for the explanation!

**Comment by parkervg** [comment]
@jixy2012 Running `pytest tests/ --ignore=tests/tasks/fact_ver_task_basic_test.py --ignore=tests/tasks/nih_basic_test.py`, confirmed tests pass with this PR

**Review by jixy2012 — APPROVED** [review]

**Pull request merged**